Exercise 3 Recursive Unhider/recursiveUnhider-rt/src/com/tridiumuniversity/recursiveUnhider/BUnhideJob.java

```diff
@@ -0,0 +1,77 @@
+package com.tridiumuniversity.recursiveUnhider;
+
+import javax.baja.job.BSimpleJob;
+import javax.baja.job.JobCancelException;
+import javax.baja.nre.annotations.NiagaraType;
+import javax.baja.sys.*;
+
+@NiagaraType
+public class BUnhideJob extends BSimpleJob
+{
+/*+ ------------ BEGIN BAJA AUTO GENERATED CODE ------------ +*/
+/*@ $com.tridiumuniversity.recursiveUnhider.BUnhideJob(2979906276)1.0$ @*/
+/* Generated Mon Sep 25 19:38:22 SGT 2023 by Slot-o-Matic (c) Tridium, Inc. 2012 */
+
+////////////////////////////////////////////////////////////////
+// Type
+////////////////////////////////////////////////////////////////
+
+  @Override
+  public Type getType() { return TYPE; }
+  public static final Type TYPE = Sys.loadType(BUnhideJob.class);
+
+/*+ ------------ END BAJA AUTO GENERATED CODE -------------- +*/
+
+  public BUnhideJob(){}
+  public BUnhideJob(BComponent rootComp){
+    this.rootComp = rootComp;
+  }
+
+  @Override
+  public void run(Context context){
+    unhideSlots(this.rootComp);
+  }
+
+  private void unhideSlots(BComponent parentComp){
+    if(parentComp == null) throw new IllegalArgumentException("Root component must be valid.");
+
+    // loop through all the slots
+    Slot[] slots = parentComp.getSlotsArray();
+
+    for(Slot slot:slots){
+      if (canceled)
+      {
+        throw new JobCancelException();
+      }
+      heartbeat();
+
+      if(Flags.isHidden(parentComp,slot)){
+        int propFlags = parentComp.getFlags(slot);
+        int newFlags = propFlags & ~Flags.HIDDEN;

```

Comment:
Doing the bit manipulation yourself is perfectly fine, but there's also a built-in method `Flags.remove()` that handles this for you.